cluster-autoscaler/cloudprovider/ionoscloud/client.go

```diff
@@ -80,80 +78,81 @@ func NewAPIClient(token, endpoint string, insecure bool) APIClient {
 		}
 	}
 	config.Debug = klog.V(6).Enabled()
+	config.SetDepth(0)
 	client := ionos.NewAPIClient(config)
 	return client.KubernetesApi
 }
 
 // AutoscalingClient is a client abstraction used for autoscaling.
 type AutoscalingClient struct {
-	client       APIClient
+	clientProvider
 	clusterId    string
-	endpoint     string
-	insecure     bool
 	pollTimeout  time.Duration
 	pollInterval time.Duration
-	tokens       map[string]string
 }
 
 // NewAutoscalingClient contructs a new autoscaling client.
-func NewAutoscalingClient(config *Config) (*AutoscalingClient, error) {
+func NewAutoscalingClient(config *Config, userAgent string) (*AutoscalingClient, error) {
 	c := &AutoscalingClient{
-		clusterId:    config.ClusterId,
-		endpoint:     config.Endpoint,
-		insecure:     config.Insecure,
-		pollTimeout:  config.PollTimeout,
-		pollInterval: config.PollInterval,
-	}
-	if config.Token != "" {
-		c.client = apiClientFactory(config.Token, config.Endpoint, config.Insecure)
-	} else if config.TokensPath != "" {
-		tokens, err := loadTokensFromFilesystem(config.TokensPath)
-		if err != nil {
-			return nil, err
-		}
-		c.tokens = tokens
```

Comment:
This was a kind of cache for the tokens read from disk. `customClientProvider.GetClient()` will read from disk on every call. Is this changed behavior intended and if yes, why?

Reply:
Yes, this is intended because the mounted tokens might change without restarting the autoscaler